### PR TITLE
Re-enable tests

### PR DESCRIPTION
**Pull request by jasonLaster**
This enables a bunch of tests as well as fixes some errors we were seeing in pause on exceptions:

### CODE REVIEW

File: src/reducers/pause.js [hunk 1]
```diff
@@ -202,7 +202,7 @@ export function pausedInEval(state: OuterState) {
   }
 
   const exception = state.pause.pause.why.exception;
-  if (!exception) {
+  if (!exception || !exception.preview) {
```
Comment (jasonLaster) [inline src/reducers/pause.js]: this was recently added and not well tested before

File: src/test/mochitest/browser_dbg-pause-exceptions.js [hunk 2]
```diff
@@ -19,26 +19,34 @@ function caughtException() {
...
-  // Test pausing on a caught Error
+  log("3. Test pausing on a caught Error");
   caughtException();
   await waitForPaused(dbg);
   assertPausedLocation(dbg);
+
+  log("3.b Test pausing in the catch statement");
+  await resume(dbg);
+  await waitForPaused(dbg);
+  assertPausedLocation(dbg);
```
Comment (jasonLaster) [inline src/test/mochitest/browser_dbg-pause-exceptions.js]: we had previously been sloppy w/ this test and had allowed `togglePauseOnExceptions` to do an implicit resume, which was creating a race condition.

File: src/test/mochitest/head.js [hunk 3]
```diff
@@ -53,7 +53,15 @@ registerCleanupFunction(() => {
...
-  info(`${msg} ${JSON.stringify(data)}`);
+  info(`${msg} ${!data ? "" : JSON.stringify(data)}`);
+}
+
+function logThreadEvents(dbg, event) {
+  const thread = dbg.toolbox.threadClient;
+
+  thread.addListener(event, function onEvent(eventName, ...args) {
+    info(`Thread event '${eventName}' fired.`);
+  });
```
Comment (jasonLaster) [inline src/test/mochitest/head.js]: a new log that is helpful for debugging

File: src/test/mochitest/head.js [hunk 4]
```diff
@@ -355,9 +371,7 @@ async function waitForPaused(dbg) {
       const paused = isPaused(dbg);
       const scope = !!getSelectedScope(state);
       const loaded = !hasLoadingObjects(state);
-      return (
-        isPaused(dbg) && getSelectedScope(state) && !hasLoadingObjects(state)
-      );
+      return paused && scope && loaded;
```
Comment (jasonLaster) [inline src/test/mochitest/head.js]: :)

File: src/test/mochitest/head.js [hunk 5]
```diff
@@ -572,8 +586,7 @@ async function stepOut(dbg) {
  * @return {Promise}
  * @static
  */
-function resume(dbg) {
-  info("Resuming");
```
Comment (codehag) [inline src/test/mochitest/head.js]: do we want to remove this comment?

File: src/test/mochitest/head.js [hunk 6]
```diff
@@ -572,8 +586,7 @@ async function stepOut(dbg) {
...
+async function resume(dbg) {
```
Comment (codehag) [inline src/test/mochitest/head.js]: does this need the async keyword? if the dbg.actions.resume() is a promise, I think you can `await` it

**Pull request merged**